### ISSUE TITLE
OpenSSL legacy provider failed to load

### ISSUE THREAD

**Issue by webzest** [openssl/openssl#19191]
Error occurred when running Nmap 7.93
Openssl version 1.1.1.1.q is installed.
Edition	Windows 11 Pro
Version	22H2
Installed on	‎9/‎6/‎2022
OS build	22622.590
Experience	Windows Feature Experience Pack 1000.22634.1000.0
--------------------------
Ran command: nmap -T4 -A -v 192.168.1.12

The following error was generated:
Starting Nmap 7.93 ( https://nmap.org ) at 2022-09-11 18:01 Eastern Daylight Time
NSOCK ERROR [0.4740s] ssl_init_helper(): OpenSSL legacy provider failed to load.

**Comment by t8m** [comment]
I'd suggest reporting this to nmap. It looks like a nmap bug. There are no providers in OpenSSL 1.1.1.

**Comment by Aholicknight** [comment]
I am getting the same error @webzest did you find a fix for this?

**Comment by webzest** [comment]
Yes, I did get an answer.  The NMap Developer suggested that I downloaded and install version 7.92.  I did and there is no more OpenSSL errors.

**Comment by webzest** [comment]
Installing version 7.92 of nMap solved the error.

**Comment by qedpro10** [comment]
I'm seeing this error with 7.93

**Comment by webzest** [comment]
@qedpro10,  Yes, version 7.93 is the issue.  Download and run v7.92 to get rid of this OpenSSL error.  It works well with Zenmap and I am running intense scans without issues.

**Comment by biva** [comment]
For reference, here is the link to download it: https://nmap.org/dist/nmap-7.92-setup.exe

**Comment by sfc9982** [comment]
I'm seeing this error with 7.93

**Comment by wan7teen-ping** [comment]
Hi I will help you with everything you need but you have to subscribe to
the youtube channel and I will fulfill all your requests

https://youtu.be/aIaqr4eq18g

Pada Isn, 12 Sep 2022, 6:09 AM Johnny Sandaire ***@***.***>
menulis:

**Comment by lucaszhang99** [comment]
Same Error on 7.93, download 7.92 will fix it

**Comment by k8scat** [comment]
<img width="602" alt="image" src="https://user-images.githubusercontent.com/55872984/198846434-e6ce6f8d-d7a3-487b-b265-f422250d1322.png">

**Comment by YangQing1989** [comment]
```
Starting Nmap 7.93 ( https://nmap.org ) at 2022-12-13 21:09 中国标准时间
NSOCK ERROR [0.0580s] ssl_init_helper(): OpenSSL legacy provider failed to load.
```
Did the bug been solved yet?

**Comment by steowens** [comment]
Ok so how do we downgrade from 7.93 back to 7.92

**Comment by webzest** [comment]
I downloaded a Kali Linux 2022.4 (debian) VM, which came with NMAP 7.93 and OpenSSL 3.0.7 already installed.  I ran it and did not get the Error.   This error may be related to the Windows OpenSSL Version compatibility with Nmap 7.93.

**Comment by jinbixiaopaigu** [comment]
> 好的，那么我们如何从 7.93 降级回 7.92

上面都有人发过下载链接

**Comment by paolobelluticm** [comment]
@steowens
> Ok so how do we downgrade from 7.93 back to 7.92

https://nmap.org/dist/nmap-7.92-setup.exe

**Comment by steowens** [comment]
Thank you!

**Comment by saadpocalypse** [comment]
7.92 works for MacOS as well. Was getting the same issue on 7.93.

**Comment by wcbutler** [comment]
Just had this issue with 7.93, doing the workaround and downgrading just noting this is still an issue.

**Comment by eabase** [comment]
Still an issue. 
*They* need to update the `Insecure.Nmap` winget package!

**Comment by jaisonora** [comment]
Still having the same issue

**Comment by saadpocalypse** [comment]
> Issue is fixed in Nmap 7.94 (tested with 1.77 npcap).

Is it fixed on a particular OS or universally? Only 7.92 works for me on MacOS.

**Comment by ku4eto** [comment]
> > Issue is fixed in Nmap 7.94 (tested with 1.77 npcap).
> 
> Is it fixed on a particular OS or universally? Only 7.92 works for me on MacOS.

Issue is for Windows. So is my reply.
